server/middlewares/authentication.js

```diff
@@ -8,69 +8,79 @@ import addMonths from 'date-fns/add_months';
 import addMinutes from 'date-fns/add_minutes';
 import { stripSubdomain } from '../../shared/utils/domains';
 
-export default function auth(options?: { required?: boolean } = {}) {
-  return async function authMiddleware(ctx: Context, next: () => Promise<*>) {
-    let token;
-
-    const authorizationHeader = ctx.request.get('authorization');
-    if (authorizationHeader) {
-      const parts = authorizationHeader.split(' ');
-      if (parts.length === 2) {
-        const scheme = parts[0];
-        const credentials = parts[1];
-
-        if (/^Bearer$/i.test(scheme)) {
-          token = credentials;
-        }
-      } else {
-        throw new AuthenticationError(
-          `Bad Authorization header format. Format is "Authorization: Bearer <token>"`
-        );
+function extractTokenFromCtx(ctx: Context): any {
```

Comment:
I'd like to stick to the "getXForY" pattern on these method names

```suggestion
function getTokenForCtx(ctx: Context): ?string {
```